api/metrics/v1/http.go

```diff
@@ -114,13 +124,34 @@ func WithWriteMiddleware(m func(http.Handler) http.Handler) HandlerOption {
 	}
 }
 
+func WithAlertmanagerAlertsReadMiddleware(m ...func(http.Handler) http.Handler) HandlerOption {
```

Comment:
As we can pass multiple
```suggestion
func WithAlertmanagerAlertsReadMiddlewares(m ...func(http.Handler) http.Handler) HandlerOption {
```

Reply:
Right, I kept the naming convention because Global already accepts multiple. I kind of think it still makes sense even without the plural for variadic arguments, wdyt?

Reply:
Ah I see! Fine to skip, this is a tiny nitpick anyway